src/Extension/Traits/FluentAdminTrait.php

```diff
@@ -566,4 +577,90 @@ protected function inEveryStage($doSomething)
             });
         }
     }
+
+    /**
+     * Update save action based on localisation state
+     * valid states:
+     * - localised (localise label)
+     * - not localised (save label)
+     *
+     * @param FieldList $actions
+     * @param DataObject $record
+     */
+    protected function updateSaveAction(FieldList $actions, DataObject $record)
+    {
+        if ($record->hasExtension(Versioned::class)) {
+            return;
+        }
+
+        if ($record->existsInLocale()) {
+            // keep the action unchanged as the record is localised
+            return;
+        }
+
+        $saveAction = $actions->fieldByName('MajorActions.action_doSave');
+
+        if (!$saveAction) {
+            return;
+        }
+
+        // update save action label to localise in case record is not localised yet
+        $saveAction
+            ->setTitle(_t(FluentExtension::class . '.Localise', 'Localise'))
```

Comment:
Use a string literal here for the FQCN. There's a few other places in this PR but I won't mention them any more.

Reply:
Yeah it's annoying but I am not sure if the text collector supports class constants yet.

Reply:
It supports `self::class` and `__CLASS__` at the moment